pytorch_lightning/trainer/trainer.py

```diff
@@ -957,6 +954,8 @@ def _run(self, model: "pl.LightningModule") -> Optional[Union[_EVALUATE_OUTPUT,
 
         # check if we should delay restoring checkpoint till later
         if not self.accelerator.restore_checkpoint_after_pre_dispatch:
+            if self._ckpt_path:
+                self._load_checkpoint_weights()
```

Comment:
note to reviewer: this must be moved after the `setup_environment` is called to ensure that torch distributed is initialized. it also matches the symmetry in later on, and makes it easier for us to reason about merging these checkpoint loading paths altogether.